.changeset/nine-mails-crash.md

```diff
@@ -0,0 +1,9 @@
+---
+'@backstage/backend-common': patch
+---
+
+The `ZipArchiveResponse` now correctly handles corrupt ZIP archives.
+
+Before this change, certain corrupt ZIP archives either cause the inflater to throw (as expected), or will hang the parser indefinitely.
+
+This change introduces a default timeout of 3000ms before throwing an error message when trying to unzip an archive.
```

Comment:
Mildly concerned about this one I guess - so there's absolutely no other way of detecting the condition from the outside? I wonder if 3s is too short, and whether this actually leaves cruft behind (like an infinite loop going on in the background, or holding on to allocated resources) that will pile up to crash the service 🤔

Reply:
My first try was to attach a listener to the stream to see if it is `.pipe()`'d but never receives any data. However by attaching a `.on('data', ...)` listener the stream actually changes reading modes (https://nodejs.org/api/stream.html#two-reading-modes) which does correctly detect broken files but also breaks the happy case. Therefore I went with the sub-optimal timed approach.

As to memory leaks, this change is exactly preventing that. Currently, if a corrupted ZIP file is being extracted, the stream will hang and remain in memory (probably along with the file buffer of the archive). Now the stream will emit an error after latest 3s, the promise will reject and the respective `ReadTreeResponse` operation will throw